std-bits/table/src/main/java/org/enso/table/format/xlsx/Range.java

```diff
@@ -0,0 +1,241 @@
+        new int[] {
+          Math.min(tl[0], br[0]),
+          Math.min(tl[1], br[1]),
+          Math.max(tl[0], br[0]),
+          Math.max(tl[1], br[1])
+        });
+  }
+
+  private static boolean isLetter(char c) {
+    return c >= 'A' && c <= 'Z';
+  }
+
+  private static boolean isDigit(char c) {
+    return c >= '0' && c <= '9';
+  }
+
+  private static int skipDollar(CharSequence address, int index) {
+    if (index < address.length() - 1 && address.charAt(index) == '$') {
+      index++;
+    }
+    return index;
+  }
+
+  private static int[] parseA1(CharSequence address) {
+    ParsedInteger col = parseColumn(address);
+    ParsedInteger row = parseInteger(address, skipDollar(address, col.index));
+    return new int[] {row.value, col.value};
+  }
+
+  private static int[] parseRC(CharSequence address) throws IllegalArgumentException {
+    int index = 0;
+
+    int row = 0;
+    if (index < address.length() && address.charAt(index) == 'R') {
+      ParsedInteger parsed = parseInteger(address, index + 1);
+      if (parsed.value == 0) {
+        throw new IllegalArgumentException(address + " not an absolute R1C1 style addresses.");
+      }
+
+      row = parsed.value;
+      index = parsed.index;
+    }
+
+    int col = 0;
+    if (index < address.length() && address.charAt(index) == 'C') {
+      ParsedInteger parsed = parseInteger(address, index + 1);
+      if (parsed.value == 0) {
+        throw new IllegalArgumentException(address + " not an absolute R1C1 style addresses.");
+      }
+
+      col = parsed.value;
+    }
+
+    return new int[] {row, col};
+  }
+
+  /**
+   * Convert an Excel Column Name (e.g. DCR) into the index (1-based)
+   *
+   * @param column name
+   * @return Column index (A=1 ...)
+   */
+  public static int parseA1Column(CharSequence column) throws IllegalArgumentException {
+    ParsedInteger parsed = parseColumn(column);
+    if (parsed.index != column.length() || parsed.value == 0) {
+      throw new IllegalArgumentException(column + " is not a valid Excel Column Name.");
+    }
+
+    return parsed.value;
+  }
+
+  private static class ParsedInteger {
+    public final int index;
+    public final int value;
+
+    public ParsedInteger(int index, int value) {
+      this.index = index;
+      this.value = value;
+    }
+  }
+
+  private static ParsedInteger parseInteger(CharSequence address, int index) {
+    int endIndex = index;
+    while (endIndex < address.length() && isDigit(address.charAt(endIndex))) {
+      endIndex++;
+    }
+    return new ParsedInteger(
+        endIndex, endIndex == index ? 0 : Integer.parseInt(address, index, endIndex, 10));
+  }
+
+  private static ParsedInteger parseColumn(CharSequence column) {
+    int col = 0;
+
+    int index = skipDollar(column, 0);
+
+    while (index < column.length() && isLetter(column.charAt(index))) {
+      col = 26 * col + (column.charAt(index) - 'A' + 1);
+      index++;
+    }
+
+    return new ParsedInteger(index, col);
+  }
```

Comment:
I find it weird that `parseColumn` takes care of `skipDollar`, but `parseInteger` does not and so it has to be done by the caller. It feels inconsistent. Can we do something about it?